core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -81,7 +81,7 @@
 
   /** Executor for application processing. */
   private Executor executor;
-  @GuardedBy("lock") private boolean usingSharedExecutor;
+  private boolean usingSharedExecutor;
```

Comment:
> I don't feel comfortable removing documentation that describes how variables should be properly synchronized.

You _just_ added it in #2011. But it was wrong. I have little issue removing wrong docs, especially as they break compilation internally.

> I don't think moving usingSharedExecutor and executor to inside the synchronized block is overkill. What bad thing is going to happen?

It is completely unnecessary. And it would be on the fast path.